### PR TITLE
Chore Update dependencies

### PR DESCRIPTION
**Pull request by wilwade**
Notes:
- This does NOT update `@polkadot/api` to v11 yet as `api-augment` does not support it yet.
- esmodules is now preferred over `namespace`

Big Updates
- Nestjs v10 (required changing to a fork of the redis package)
- Eslint v9

### CODE REVIEW

File: eslint.config.mjs
Comment (wilwade) [inline eslint.config.mjs:1]: This does remove a few lints that do not yet support v9.

File: setup/testing/index.ts
Comment (wilwade) [inline setup/testing/index.ts:1]: `ItemizedSignaturePayloadV2` is not available in the `@amplica-labs/frequency-scenario-template`. @JoeCap08055 perhaps you know why? But I don't think it matters for the interface usage.

File: setup/testing/index.ts [hunk 1]
```diff
@@ -147,8 +143,8 @@ async function main() {
   log.setLevel('trace');
   const { apiPromise } = ExtrinsicHelper;
 
-  const provider: ChainUser = { keys: keyring.addFromUri('//Alice') };
-  const famousUser: ChainUser = { uri: '//Bob', keys: keyring.addFromUri('//Bob') };
+  const provider: ChainUser = { keys: keyring.createFromUri('//Alice') };
+  const famousUser: ChainUser = { uri: '//Bob', keys: keyring.createFromUri('//Bob') };
```
Comment (wilwade) [inline setup/testing/index.ts:147]: Better function for how these are used.

File: setup/testing/index.ts [hunk 2]
```diff
@@ -168,10 +164,10 @@ async function main() {
   // Create all keypairs
   console.log('Creating keypairs...');
   new Array(7000).fill(0).forEach((_, index) => {
-    const keys = keyring.addFromUri(`//Charlie//${index}`);
+    const keys = keyring.createFromUri(`//Charlie//${index}`);
     followers.set(keys.address, { uri: `//Charlie//${index}`, keys, resetGraph: [] });
   });
-  const [followerAddresses] = mapToArray(followers);
+  const followerAddresses = [...followers.keys()];
```
Comment (wilwade) [inline setup/testing/index.ts:170]: Cleanup

File: setup/testing/index.ts [hunk 3]
```diff
@@ -374,7 +370,7 @@ User graphs to clear: ${graphsToClear}
   };
 
   const responses: ProviderResponse[] = [];
-  const [_, allFollowers] = mapToArray(followers);
+  const allFollowers = [...followers.values()]
```
Comment (wilwade) [inline setup/testing/index.ts:373]: Cleanup

File: src/blockchain-scanner.service.ts [hunk 4]
```diff
@@ -1,9 +1,10 @@
+import '@frequency-chain/api-augment';
 import { Logger, OnApplicationBootstrap } from '@nestjs/common';
 import { BlockHash } from '@polkadot/types/interfaces';
 import { OnQueueEvent, QueueEventsHost, QueueEventsListener } from '@nestjs/bullmq';
 import { BlockchainService } from './blockchain/blockchain.service';
 import { ReconnectionCacheMgrService } from './cache/reconnection-cache-mgr.service';
-import { ReconnectionServiceConstants } from './constants';
+import * as ReconnectionServiceConstants from './constants';
```
Comment (wilwade) [inline src/blockchain-scanner.service.ts:7]: Namespaces are now considered bad form and to use this pattern instead.

File: src/blockchain/extrinsic.ts [hunk 5]
```diff
@@ -34,7 +34,7 @@ import { KeyringPair } from '@polkadot/keyring/types';
 import { timeout } from 'rxjs/operators';
 import { EventError } from './event-error';
 
-export type EventMap = { [key: string]: Event };
+export type EventMap = Record<string, Event>;
```
Comment (wilwade) [inline src/blockchain/extrinsic.ts:37]: Auto from linting

File: src/graph/graph-state-manager.ts
Comment (wilwade) [inline src/graph/graph-state-manager.ts:1]: Linting:
- Unused imports and lint pattern

File: tsconfig.json
Comment (wilwade) [inline tsconfig.json:1]: Mostly formatting

File: tsconfig.json [hunk 6]
```diff
@@ -1,39 +1,30 @@
...
+    "strict": true,
+    "skipLibCheck": true,
+    "strictPropertyInitialization": false,
+    "target": "es2022",
+    "typeRoots": ["node_modules/@types"],
+    "noEmit": false,
+    "traceResolution": true
+  },
+  "include": ["./**/*.ts"],
+  "exclude": ["node_modules/**", "./dist/**", "/tools/**", "./setup/**", "./**/*.spec.ts", "./jest.init.ts"]
```
Comment (wilwade) [inline tsconfig.json:29]: Change: Notice we are now excluding the jest and spec files

File: jest.init.ts
Comment (wilwade) [inline jest.init.ts:1]: Make sure jest uses api-augment

File: webhook-specification/mock-webhook-server/package.json [hunk 7]
```diff
@@ -3,20 +3,19 @@
   "version": "1.0.0",
   "description": "",
   "scripts": {
-    "start": "ts-node-dev -r tsconfig-paths/register src/main.ts"
+    "start": "tsx src/main.ts"
   },
   "keywords": [],
   "author": "",
-  "license": "ISC",
+  "license": "Apache-2.0",
```
Comment (wilwade) [inline webhook-specification/mock-webhook-server/package.json:10]: Fixed license

File: webhook-specification/mock-webhook-server/package.json [hunk 8]
```diff
@@ -3,20 +3,19 @@
   "version": "1.0.0",
   "description": "",
   "scripts": {
-    "start": "ts-node-dev -r tsconfig-paths/register src/main.ts"
+    "start": "tsx src/main.ts"
```
Comment (wilwade) [inline webhook-specification/mock-webhook-server/package.json:6]: Swap to the better tsx

File: eslint.config.mjs
Comment (wilwade) [inline eslint.config.mjs:1]: This does remove a few lints that do not yet support v9.

File: setup/testing/index.ts
Comment (saraswatpuneet) [reply setup/testing/index.ts:1]: I see many packages were removed, did you do it manually or is there some npm trick ?

Comment (wilwade) [reply setup/testing/index.ts:1]: Manually :(

Comment (JoeCap08055) [reply setup/testing/index.ts:1]: I think I went down a rabbit hole one day looking for `ItemizedSignaturePayloadV2` and couldn't figure out the issue

File: src/blockchain-scanner.service.ts (hunk 4 above)
Comment (JoeCap08055) [reply src/blockchain-scanner.service.ts:7]: Yeah, I've noticed that recently. Not sure if I agree with the thinking there--I think the name of the namespace should be determined by the author, not the importer. But that's a philosophical debate...

File: webhook-specification/mock-webhook-server/package.json (hunk 8 above)
Comment (wilwade) [reply webhook-specification/mock-webhook-server/package.json:6]: 10-4. I'll roll that change back. It appeared to work, but I wasn't sure

Comment (wilwade) [reply webhook-specification/mock-webhook-server/package.json]: Reverted.

**Review by JoeCap08055 — APPROVED** [review]
Added a commit to fix the mock webhook server under Docker.
Approved.

File: package.json [hunk 9]
```diff
@@ -34,63 +34,56 @@
...
+    "@nestjs/schedule": "^4.0.2",
+    "@nestjs/testing": "^10.3.8",
+    "@nestjs/typeorm": "^10.0.2",
+    "@polkadot/api": "^10.12.4",
+    "@polkadot/api-base": "^10.12.4",
+    "@polkadot/keyring": "^12.6.2",
+    "@polkadot/types": "^10.12.4",
+    "@polkadot/util": "^12.6.2",
+    "@polkadot/util-crypto": "^12.6.2",
+    "@songkeys/nestjs-redis": "^10.0.0",
```
Comment (saraswatpuneet) [inline package.json:60]: any reason to switch to this flavor or redis?

File: setup/testing/index.ts
Comment (mattheworris) [reply setup/testing/index.ts:1]: I recently used `depcheck` on account-service to find unused dependencies, works pretty well.

**Review by mattheworris — APPROVED** [review]
- Read through changes
lgtm

**Pull request merged**